lib/registrant.rb

```diff
@@ -0,0 +1,27 @@
+class Registrant
+
+    attr_reader :name, :age, :license_data
+
+    def initialize(name, age,permit = false)
+        @name = name
+        @age = age
+        @permit = permit
+        @license_data = {
+            written: false,
+            license: false,
+            renewed: false
+        }
+    end
+
+    def permit?
+        @permit
+    end
+    
+    def earn_permit
+        @permit = true
```

Comment:
This is great. You could also add a what is called a guard clause to ensure that a permit cannot be earned if a registrant is not eligible.
```ruby
@permit = true if @age >= 15
```